docs/modules/qbi-helper/examples.md

```diff
@@ -0,0 +1,124 @@
+
+* players within a 20m radius
+* minimap objects within a 50m x 50m box
+
+This translates to two queries:
+
+* `Position` and `PlayerInfo` component updates for entities:
+  * with a `PlayerInfo` component
+  * are within a 20m radius of the client's player
+* `Position` and `MinimapRepresentation` component updates for entities:
+  * with a `MinimapRepresentation` component
+  * are within a 50m x 50m box around the client's player
+
+We can build up our [`Constraint`]({{urlRoot}}/api/query-based-interest/constraint) and use this to construct our [`InterestQuery`]({{urlRoot}}/api/query-based-interest/interest-query):
+
+```csharp
+var playerQuery = InterestQuery
+    .Query(
+        Constraint.All(
+            Constraint.Component<PlayerInfo.Component>(),
+            Constraint.RelativeSphere(20)))
+    .FilterResults(Position.ComponentId, PlayerInfo.ComponentId);
+
+var minimapQuery = InterestQuery
+    .Query(
+        Constraint.All(
+            Constraint.Component<MinimapRepresentation.Component>(),
+            Constraint.RelativeBox(50, double.PositiveInfinity, 50)))
+    .FilterResults(Position.ComponentId, MinimapRepresentation.ComponentId);
+```
+
+We can then use the [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template) class to construct our interest. As we are specifying interest for the client-worker, we tie the queries to a component the client-worker is authoritative over, `PlayerControls`:
+
+```csharp
+var interestTemplate = InterestTemplate.Create()
+    .AddQueries<PlayerControls.Component>(playerQuery, minimapQuery);
+```
+
+Lastly, we add the interest component to our player entity template.
+
+```csharp
+entityTemplate.AddComponent(interestTemplate.ToSnapshot(), WorkerUtils.UnityGameLogic);
+```
+
+The worker authoritative over the `Interest` component can make changes to the interest queries at runtime. For example, you may decide to remove the minimap query if the player disables the minimap.
+
+## Teams
+
+This example shows how a player could observe the position of all other players on their team.
+
+Suppose there is a Red team and a Blue team. Entities representing players are given either a `RedTeam` or `BlueTeam` component by a server-worker, to express which team they belong to.
+
+We then consider two components:
+
+* `PlayerControls`, authoritative on client-worker
+* `RedTeam` or `BlueTeam`, authoritative on server-worker
+
+<%(#Expandable title="See schema")%>
+
+```
+component PlayerControls {
+   id = 2001;
+   int32 input_value = 1;
+}
+
+component RedTeam {
+   id = 2004;
+}
+
+component BlueTeam {
+   id = 2005;
+}
+```
+
+<%(/Expandable)%>
+
+Our client-worker wants to know the positions of all players with the same team component as the client-worker's player. This is represented as a query with a component constraint on either the `RedTeam` or `BlueTeam` component ID, returning just the `Position` component.
+
+```csharp
+var teamComponentId;
```

Comment:
Maybe have a stub of a method `var teamComponentId = GetPlayerTeamId();` or similar